packages/in-app-notification-manager/src/routers/notificationRouter.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  authRole,
+  ExpressContext,
+  fromAppContext,
+  validateAuthorization,
+  ZodiosContext,
+  zodiosValidationErrorToApiProblem,
+} from "pagopa-interop-commons";
+import { ZodiosRouter } from "@zodios/express";
+import { ZodiosEndpointDefinitions } from "@zodios/core";
+import { emptyErrorMapper } from "pagopa-interop-models";
+import { inAppNotificationApi } from "pagopa-interop-api-clients";
+import { InAppNotificationService } from "../services/inAppNotificationService.js";
+import { makeApiProblem } from "../model/errors.js";
+import { notificationToApiNotification } from "../model/apiConverter.js";
+import {
+  markNotificationAsReadErrorMapper,
+  deleteNotificationErrorMapper,
+} from "../utilities/errorMappers.js";
+
+export const notificationRouter = (
+  zodiosCtx: ZodiosContext,
+  service: InAppNotificationService
+): ZodiosRouter<ZodiosEndpointDefinitions, ExpressContext> => {
+  const notificationRouter = zodiosCtx.router(
+    inAppNotificationApi.notificationApi.api,
+    {
+      validationErrorHandler: zodiosValidationErrorToApiProblem,
+    }
+  );
+  const { ADMIN_ROLE, API_ROLE } = authRole;
+
+  notificationRouter
+    .get("/notifications", async function (req, res) {
+      const ctx = fromAppContext(req.ctx);
+      try {
+        validateAuthorization(ctx, [ADMIN_ROLE, API_ROLE]);
+
+        const { limit, offset, q } = req.query;
+        const { results, totalCount } = await service.getNotifications(
+          q,
+          limit,
+          offset,
+          ctx
+        );
+        return res.status(200).send({
+          results: results.map(notificationToApiNotification),
+          totalCount,
+        });
```

Comment:
I would add the return model parsing `inAppNotificationApi.Notifications.parse(...)`